worker/changestream/stream.go

```diff
@@ -0,0 +1,177 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package changestream
+
+import (
+	"database/sql"
+	"time"
+
+	"github.com/juju/clock"
+	"github.com/juju/errors"
+	"github.com/juju/juju/core/changestream"
+	"github.com/juju/juju/database"
+	"gopkg.in/tomb.v2"
+)
+
+const (
+	// PollInterval is the amount of time to wait between polling the database
+	// for new stream events.
+	PollInterval = time.Millisecond * 100
```

Comment:
I think how this will work is that it will be configurable per model, and if zero will halt the stream.
We would then add a model config watcher to the stream loop, so that we can read it from the DB for our namespace and adjust accordingly.